src/watchman_client.js

```diff
@@ -43,6 +43,12 @@ function WatchmanClient(watchmanBinaryPath) {
   this._backoffTimes = this._setupBackoffTimes();
 
   this._clientListeners = null; // direct listeners from here to watchman.Client.
+
+  // Define a handler for if somehow the Node process gets interrupted. We need to
+  // close down the watchman.Client, if we have one.
+  process.on('SIGINT', () => {
```

Comment:
Two issues:

* This can leak memory. If a `sane` session using watchman is no longer being used, it will be retained by the `process.on('SIGINT'` handler until the process shutdown (or SIGINT) is worked
* This wont handle other forms of shutdown, such as the process simply ending when it has no more work to do.

Ideas to resolve:

* the global cleanup handler needs to be disabled via `off` during `WatchmanWatcher.prototype.close`
* use [capture-exit](https://github.com/ember-cli/capture-exit) to handle all reasonable forms of exit + cooperation with other libraries attempting to do the same.